addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2138,7 +2139,30 @@ export class OdooEditor extends EventTarget {
         for (const child of [...container.childNodes]) {
             this._cleanForPaste(child);
         }
-        return container.innerHTML;
+        // All ChildnNodes at the root of the clipboardData should be blocks.
+        // This is to ensure that the HTML insertion done in `insert()`
+        // function as intended.
```

Comment:
This is not true as far as I know. Insert works fine. It's the other functions of the editor that tends to do unexpected stuff.

Reply:
Anyway, the text is still not convincing enough IMO. I suggest the following (break lines appropriately):
```
Force inline nodes at the root of the container into P elements. This is a tradeoff to ensure some features that rely on nodes having a parent (e.g. convert to list, title, etc.) can work properly on such nodes without having to actually handle that particular case in all of those functions. In fact, this case cannot happen on a new document created using this editor, but will happen instantly when editing a document that was created from Etherpad.
```

Reply:
Wait... I just realized there are two different blocks of code doing the same conceptual thing. Why are there two functions looping on nodes and putting them inside P's ? Shouldn't they both call the same function ? Something like `wrapRootInlines(container)` ? What am I missing ?

Reply:
After IRL discussion :

Too different to be elegantly merged into a single function.